trading/models.py

```diff
@@ -38,15 +38,17 @@ def available_cash(self):
         ])
         return stock_val + bucket_val
 
-    @staticmethod
-    def available_buckets(bkt):
+    def available_buckets(self, bkt):
         """
         Find the available buckets that have quantity > 0
         """
-        bucket = TradeBucket.buckettrades.filter(bucket=bkt).annotate(
+        bucket = self.buckettrades.all().filter(stock=bkt.id).annotate(
             sum_quantity=models.Sum('quantity'))
-        if bucket:
-            return bucket[0].sum_quantity
+        total_sum = 0
+        for x in bucket:
+            total_sum += x.sum_quantity
+        if total_sum is not 0:
+            return total_sum
```

Comment:
These two lines are not necessary, but you should not need these 5 lines after you addresse the comments

Reply:
unfortunately the filter/sum combo doesn't actually compute the sum across all of them... so this step is necessary :(